features/csharp/csharp_config.feature

```diff
@@ -1,165 +1,139 @@
+    And the exception "message" equals "MarkLaunchComplete"
+    And the event "app.isLaunching" is false
+  
+  # @skip_macos
+  # Scenario: Set long launch time
+  #   When I run the game in the "LongLaunchTime" state
+  #   And I wait to receive 2 errors
+  #   And I sort the errors by the payload field "events.0.exceptions.0.message"
+  #   Then the error is valid for the error reporting API sent by the Unity notifier
+  #   And the exception "message" equals "Error 1"
+  #   And the event "app.isLaunching" is true
+  #   And I discard the oldest error
+  #   And the exception "message" equals "Error 2"
+  #   And the event "app.isLaunching" is false
+  
+  # @skip_macos
+  # Scenario: Set short launch time
+  #   When I run the game in the "ShortLaunchTime" state
+  #   And I wait to receive 2 errors
+  #   And I sort the errors by the payload field "events.0.exceptions.0.message"
+  #   Then the error is valid for the error reporting API sent by the Unity notifier
+  #   And the exception "message" equals "Error 1"
+  #   And the event "app.isLaunching" is true
+  #   And I discard the oldest error
+  #   And the exception "message" equals "Error 2"
+  #   And the event "app.isLaunching" is false
+
+
+#END SECTION -----------------------------------------------------------------------------
+
+  Scenario: Auto notify false
+    When I run the game in the "AutoNotifyFalse" state
+    And I should receive no errors
+
+  Scenario: Release Stage
+    When I run the game in the "ReleaseStage" state
+    And I should receive no errors
+
+  Scenario: Set Uncaught As Unhandled
+    When I run the game in the "SetUncaughtAsUnhandled" state
     And I wait to receive an error
     Then the error is valid for the error reporting API sent by the Unity notifier
-    And the event "exceptions.0.message" equals "Outer"
-    And the event "exceptions.1.message" equals "Inner"
-    And the event "exceptions.2" is null
+    And the exception "message" equals "SetUncaughtAsUnhandled"
+    And the event "unhandled" is true
 
-  Scenario: Reporting an uncaught exception in an async method
-    When I run the game in the "AsyncException" state
+  Scenario: Discard Error Class
+    When I run the game in the "DiscardErrorClass" state
     And I wait to receive an error
     Then the error is valid for the error reporting API sent by the Unity notifier
-    And the exception "message" equals "AsyncException"
+    And the exception "message" equals "Error 2"
 
-  @skip_webgl
-  Scenario: Report exception from background thread
-    When I run the game in the "BackgroundThreadException" state
+  Scenario: App Type
+    When I run the game in the "AppType" state
     And I wait to receive an error
     Then the error is valid for the error reporting API sent by the Unity notifier
-    And the exception "message" equals "BackgroundThreadException"
+    And the exception "message" equals "AppType"
+    And the event "app.type" equals "custom"
```

Comment:
Do AppType and Context really need to be in separate scenarios?

Reply:
i'll consider merging them in the next PR 👍